py2many/inference.py

```diff
@@ -250,17 +250,20 @@ def typename(e):
     def visit_Assign(self, node: ast.Assign) -> ast.AST:
         self.generic_visit(node)
 
-        target = node.targets[0]
         annotation = getattr(node.value, "annotation", None)
-        target_has_annotation = hasattr(target, "annotation")
-        inferred = (
-            getattr(target.annotation, "inferred", False)
-            if target_has_annotation
-            else False
-        )
-        if annotation is not None and (not target_has_annotation or inferred):
-            target.annotation = annotation
-            target.annotation.inferred = True
+        if annotation is None:
+            return node
+
+        for target in node.targets:
+            target_has_annotation = hasattr(target, "annotation")
+            inferred = (
+                getattr(target.annotation, "inferred", False)
+                if target_has_annotation
+                else False
+            )
+            if not target_has_annotation or inferred:
+                target.annotation = annotation
```

Comment:
Fails on C++ and Dart (working on it), and Julia fails because tuples are implemented as `Pair` so can only have two items - do we have an issue about that?

Reply:
I've created issue https://github.com/adsharma/py2many/issues/372 .

At the moment this is no problem here identified by your case.
Tuples are not handled at all in this transformer.  https://github.com/adsharma/py2many/issues/373 is a starter on the dragons that be here, and is far better than a comment in this unrelated bit of code.

Reply:
> Fails on C++ and Dart (working on it), and Julia fails because tuples are implemented as `Pair` so can only have two items - do we have an issue about that?

I think you mean Kotlin, which doesn't have tuples:

https://discuss.kotlinlang.org/t/tuples-deprecated/787/2